### PR TITLE
[node] Add dns.setDefaultResultOrder().

### PR DESCRIPTION
**Pull request by westy92**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://nodejs.org/api/dns.html#dnssetdefaultresultorderorder
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

Fixes https://github.com/DefinitelyTyped/DefinitelyTyped/discussions/55904.

### CODE REVIEW

File: types/node/dns/promises.d.ts [hunk 1]
```diff
@@ -331,6 +331,17 @@ declare module 'dns/promises' {
...
+     * Set the default value of `verbatim` in {@link lookup}. The value could be:
+     * - `ipv4first`: sets default `verbatim` `false`.
+     * - `verbatim`: sets default `verbatim` `true`.
+     *
+     * The default is `ipv4first` and {@link setDefaultResultOrder} have higher priority than `--dns-result-order`.
+     * When using worker threads, {@link setDefaultResultOrder} from the main thread won't affect the default dns orders in workers.
+     * @since v14.18.0
+     * @param order must be 'ipv4first' or 'verbatim'.
+     */
+    function setDefaultResultOrder(order: 'ipv4first' | 'verbatim'): Promise<void>;
```
Comment (LinusU) [inline types/node/dns/promises.d.ts:344]: I cannot find anything in the documentation about this returning a `Promise<void>`. Are you sure that it doesn't just return `void`?

Comment (westy92) [reply types/node/dns/promises.d.ts]: Good catch, fixed!

**Comment by peterblazejewicz** [comment]
@westy92 please update from master and push changes, there were changes in other DT package that require update, so Node ecosystem tests can pass, thx!

**Comment by westy92** [comment]
@peterblazejewicz weird, I've tried that a few times now with no luck. :(

**Comment by westy92** [comment]
@peterblazejewicz the same issue seems to be happening here: https://github.com/DefinitelyTyped/DefinitelyTyped/pull/57224#issuecomment-971903988

**Comment by peterblazejewicz** [comment]
fixing CI resolutions: #57241

**Review by SimonSchick — APPROVED** [review]

**Pull request merged**